save_files/lib.js

```diff
@@ -192,30 +203,44 @@ function bottle_updates(tavern, vendor) {
     return updates;
 }
 
+let refill_updates = function(v) {
+    let refill = v ? 0xA0 : 0x0;
+    return[[ADDRESSES['heart_refill'], v]];
+}
+
+let mushroom_updates = function(v) {
+    if(v === 'mushroom') {
+        return [[ADDRESSES['mushroom'], 1]];
+    } else if( v === 'powder') {
+        return [[ADDRESSES['mushroom'], 2]];
+    } else {
+        return [[ADDRESSES['mushroom'], 0]];
+    }
+}
 
 function form_to_updates(form) {
     let fields = [
         ['arrows', new InputToStuff(
-            numeric, a => a >= 0 && a <= 15, address_to_value('arrows')
```

Comment:
oh i messed this up, will need to either de-genericize or find a better way to do so

can probably set min/max attrs on the inputs themselves and crib off of those here